ipn/ipnlocal/local.go

```diff
@@ -4759,3 +4773,19 @@ func (b *LocalBackend) ListProfiles() []ipn.LoginProfile {
 	defer b.mu.Unlock()
 	return b.pm.Profiles()
 }
+
+// ResetAuth resets the authentication state, including persisted keys. The
+// backend is left with a new profile, ready for StartLoginInterative to be
+// called to register it as new node.
+func (b *LocalBackend) ResetAuth() error {
+	b.mu.Lock()
+	if err := b.clearMachineKeyLocked(); err != nil {
```

Comment:
don't we have to stop some things before we do this?

like stop the control client if it's connected?

it works for your use case (already broken), but I don't want call this on a running state to crash the backend and I'm not convinced the machine key won't be accessed while it's running. not a data race, but an assumption that it can access the machine key under a lock and then transform it to a public key and have that operation crash.

Reply:
I think you also need to update `createGetMachinePrivateKeyFunc`

Reply:
`resetForProfileChangeLockedOnEntry()` ends up doing this shortly after (it calls `Start`, which calls `resetControlClientLockedAsync`). I've added an explicit call before we clear the machine key.

Reply:
> I think you also need to update `createGetMachinePrivateKeyFunc`

I don't think that's necessary (empirically a new machine key is generated). We call it again when creating the new control client, thus we end up capturing a new `cache` variable in the closure.